packages/cli/src/cmds/agentInstaller/projectConfiguration.ts

```diff
@@ -81,12 +86,47 @@ async function resolveSelectedInstallers(
     }
   }
 
-  // Sanity check. Verify everything resolved properly. This should branch should never occur.
+  // Sanity check. Verify everything resolved properly. This branch should never occur.
   if (projects.some(({ selectedInstaller }) => !selectedInstaller)) {
     throw new Error('Invalid selection');
   }
 }
 
+export async function getYarnPackages(availableInstallers: AgentInstaller[]): Promise<string[]> {
+  let yarnPackages: string[] = [];
+  for (const installer of availableInstallers) {
+    if (installer.name === 'yarn') {
+      const packageJsonFilename = join(installer.path, 'package.json');
+      if (await exists(packageJsonFilename)) {
```

Comment:
I think the recommended idiom is to try to read and rescue the error instead; the file can still disappear after `exists` so it doesn't do much good to even bother calling it.

Reply:
`exists` is used in 14 other locations in the code.

Reply:
Changed `exists` to `try/catch`.